scylla-macros/src/parser.rs

```diff
@@ -35,6 +35,46 @@ pub(crate) fn parse_named_fields<'a>(
     }
 }
 
+pub(crate) enum StructFields<'a> {
+    Named(&'a FieldsNamed),
+    Unnamed(&'a FieldsUnnamed),
+}
+
+/// Parses the tokens_input to a DeriveInput and returns the struct name from which it derives and
+/// the fields
+pub(crate) fn parse_struct_fields<'a>(
+    input: &'a DeriveInput,
+    current_derive: &str,
+) -> Result<StructFields<'a>, syn::Error> {
+    match &input.data {
+        Data::Struct(data) => match &data.fields {
+            Fields::Named(named_fields) => Ok(StructFields::Named(named_fields)),
+            Fields::Unnamed(unnamed_fields) => Ok(StructFields::Unnamed(unnamed_fields)),
+            Fields::Unit => Err(syn::Error::new_spanned(
+                data.struct_token,
+                format!(
+                    "derive({}) works only for structs with fields",
+                    current_derive
+                ),
```

Comment:
Maybe extract this to a function?

Reply:
What exactly do you suggest to extract?

Reply:
The whole `format!` expression, parametrised by `current_derive`.

Reply:
Extracted it to the closure. Did the same for `parse_named_fields` utility function.